File: java/src/ev1/ut2_controlflujometodos/u04metodos/teoria/U05_MetodosEstructurasIntro.java
package ev1.ut2_controlflujometodos.u04metodos.teoria;

/**
 * Teoria: metodos que usan estructuras de control (if, bucles, switch).
 *
 * <p><b>Objetivo:</b> combinar metodos con condicionales y bucles para resolver problemas mas completos.</p>
 *
 * <p>Dentro de un metodo puedes usar las mismas estructuras que en {@code main}:</p>
 * <ul>
 *   <li><b>if / else</b> - devolver distintos resultados segun una condicion.</li>
 *   <li><b>for / while / do-while</b> - repetir calculos (sumar 1..n, tablas, contadores).</li>
 *   <li><b>switch</b> - elegir un resultado entre varias opciones fijas (dia de la semana, menu).</li>
 * </ul>
 *
 * <p>La ventaja de meter la logica en un metodo es que {@code main} queda limpio: lee datos,
 * llama al metodo adecuado e imprime el resultado.</p>
 *
 * <pre>{@code
 * public static String calificarNota(double nota) {
 *     if (nota >= 5) {
 *         return "Aprobado";
 *     }
 *     return "Suspenso";
 * }
 * }</pre>
 *
 * <p>Equivale a los Ejercicios 7-10 en {@code ejercicios/}.</p>
 *
 * @author Agustin. A. Marquez. Pina
 * @since 03/06/2026
 * @see <a href="mailto:agu1406@outlook.es">agu1406@outlook.es</a>
 * @see <a href="https://github.com/Agu1406/ClasesParticulares">Repositorio GitHub</a>
 * @see <a href="https://www.agustinmarquez.dev">Sitio web</a>
 */
public class U05_MetodosEstructurasIntro {

    public static String calificarNota(double nota) {
        if (nota >= 5) {
            return "Aprobado";
        }
        return "Suspenso";
    }

    public static int sumarDelUnoAlN(int limite) {
        int suma;
        int indice;
        suma = 0;
        for (indice = 1; indice <= limite; indice++) {
            suma += indice;
        }
        return suma;
    }

    public static String nombreDia(int dia) {
        switch (dia) {
            case 1:
                return "Lunes";
            case 2:
                return "Martes";
            case 3:
                return "Miercoles";
            case 4:
                return "Jueves";
            case 5:
                return "Viernes";
            case 6:
                return "Sabado";
            case 7:
                return "Domingo";
            default:
                return "Invalido";
        }
    }

    public static String resumenNumero(int numero) {
        String paridad;
        int suma;
        int indice;
        paridad = (numero % 2 == 0) ? "par" : "impar";
        suma = 0;
        for (indice = 1; indice <= numero; indice++) {
            suma += indice;
        }
        return "Numero " + numero + ", " + paridad + ", suma 1..n = " + suma;
    }

    public static void main(String[] args) {
        System.out.println(calificarNota(6.5));
        System.out.println(calificarNota(4.0));
        System.out.println("Suma 1..10: " + sumarDelUnoAlN(10));
        System.out.println(nombreDia(3));
        System.out.println(resumenNumero(8));
    }
}

File: csharp/src/ev1/ut2_controlflujometodos/u04metodos/teoria/U05_MetodosEstructurasIntro.cs
/*
U05 — Metodos que combinan if, bucles y return.

OBJETIVO:
  - Escribir metodos que usen estructuras de control internamente.
  - Mezclar condicionales y bucles dentro de un metodo reutilizable.
  - Devolver resultados calculados tras procesar datos.

Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/

class Program
{
    static void Main()
    {
        Console.WriteLine("\n¡INICIO DEL PROGRAMA!\n");

        /*
        PRIMERA PARTE - Metodo con if/else if para clasificar.
        */
        Console.WriteLine("¡DEMO 1: CLASIFICAR NOTA!\n");
        ImprimirResultado(9);
        ImprimirResultado(6);
        ImprimirResultado(3);

        /*
        SEGUNDA PARTE - Metodo con bucle for que acumula.
        */
        Console.WriteLine("\n¡DEMO 2: SUMA DEL 1 AL N!\n");
        Console.WriteLine($"Suma 1..10 = {SumarHasta(10)}");
        Console.WriteLine($"Suma 1..100 = {SumarHasta(100)}");

        /*
        TERCERA PARTE - Metodo con if + for: contar pares en un rango.
        */
        Console.WriteLine("\n¡DEMO 3: CONTAR PARES!\n");
        Console.WriteLine($"Pares entre 1 y 20: {ContarPares(1, 20)}");
        Console.WriteLine($"Pares entre 10 y 15: {ContarPares(10, 15)}");

        /*
        CUARTA PARTE - Metodo con bucle + if + return: buscar multiplo.
        */
        Console.WriteLine("\n¡DEMO 4: PRIMER MULTIPLO!\n");
        Console.WriteLine($"Primer multiplo de 7 >= 50: {PrimerMultiplo(7, 50)}");

        Console.WriteLine("\n¡FIN DEL PROGRAMA!\n");
    }

    static void ImprimirResultado(int nota)
    {
        string texto;
        if (nota >= 9)
        {
            texto = "Sobresaliente";
        }
        else if (nota >= 5)
        {
            texto = "Aprobado";
        }
        else
        {
            texto = "Suspenso";
        }

        Console.WriteLine($"  Nota {nota} -> {texto}");
    }

    static int SumarHasta(int n)
    {
        int total = 0;
        for (int i = 1; i <= n; i++)
        {
            total += i;
        }
        return total;
    }

    static int ContarPares(int desde, int hasta)
    {
        int contador = 0;
        for (int i = desde; i <= hasta; i++)
        {
            if (i % 2 == 0)
            {
                contador++;
            }
        }
        return contador;
    }

    static int PrimerMultiplo(int baseNum, int minimo)
    {
        int candidato = minimo;
        while (candidato % baseNum != 0)
        {
            candidato++;
        }
        return candidato;
    }
}
